src/frontend/src/pages/MyGroup/PendingApplications.jsx

```diff
@@ -1,33 +1,93 @@
-import { Table, TableBody, TableContainer } from '@carbon/react';
-
-import { SmsDataTable, SimpleHeader, SimpleRow } from 'components/SmsDataTable';
+import {
+    Table,
+    TableBody,
+    TableContainer,
+    TableCell,
+    TableRow,
+    TableBatchActions,
+    TableBatchAction,
+    TableToolbar,
+    TableSelectRow,
+    TableHeader,
+    TableHead,
+    TableSelectAll
+} from '@carbon/react';
+import { Close } from '@carbon/react/icons';
+import { SmsDataTable } from 'components/SmsDataTable';
+import { useManageGroupApplication } from 'hooks';
 
 export function PendingApplications(groupApplications) {
+    const { rejectGroupApplication } = useManageGroupApplication();
     const headers = [
         { key: 'name', header: 'Name', isSortable: true },
         { key: 'message', header: 'Message' }
-        // TODO: add management logic
-        // { key: 'options', header: 'Options' }
     ];
     const applications = groupApplications.groupApplications;
 
+    const batchActionClick = selectedRows => () => {
+        rejectGroupApplication({
+            ids: selectedRows.filter(data => data.hasOwnProperty('id')).map(data => data.id)
+        });
+        window.location.reload(false);
```

Comment:
this is cool, I meant to include this in previous changes. However, it may have unintended effects, as click the "reject" button will now close the modal (on refresh). This isn't great UX.

Instead, a refresh callback when the modal closes would do what we want.

_edit_ CBS makes this fun, easy, and simple by **not providing an onClose** handler. :/ So, we would need to hack our own somehow... Let's just leave this out for now and make a comment